src/main/java/courseManager/Course.java

```diff
@@ -0,0 +1,109 @@
+package courseManager;
+
+import java.util.ArrayList;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+
+public class Course {
+    private String courseCode;
+    private String courseName;
+    private String semester;
+    private boolean archived;
+    private float credit;
+
+    private Outline outline;
+
+    private ArrayList<CourseEvent> courseEvents;
+
+    /**
+     * Create a new Course object with the given parameters
+     * @param courseCode the code of the course
+     *                   e.g. 'CSC207'
+     * @param courseName the name of the course
+     *                   e.g. 'Software Design'
+     * @param semester the semester the course is offered in
+     *                   e.g. 'Fall 2022'
+     * @param credit the number of credits the course is worth; must be either 0.5 or 1.0
+     * @param archived whether the course is archived or not
+     */
+
+    public Course(String courseCode, String courseName, String semester, boolean archived, float credit) {
+        this.courseCode = courseCode;
+        this.courseName = courseName;
+        this.semester = semester;
+        this.archived = archived;
+        this.credit = credit;
+        this.courseEvents = new ArrayList<>();
+    }
+
+    public void setOutline(Outline outline) {
+        this.outline = outline;
+    }
+
+    public String getCourseCode() {
+        return courseCode;
+    }
+
+    public String getCourseName() {
+        return courseName;
+    }
+
+    public String getSemester() {
+        return semester;
+    }
+
+    public boolean isArchived() {
+        return archived;
+    }
+
+    public float getCredit() {
+        return credit;
+    }
+
+    public Outline getOutline() {
+        return outline;
+    }
+
+    public ArrayList<CourseEvent> getCourseEvents() {
+        return courseEvents;
+    }
+
+    public void addCourseEvent(CourseEvent courseEvent) {
+        if (courseEvents.contains(courseEvent)) {
+            throw new IllegalArgumentException("Course event already exists");
```

Comment:
Since you are intentionally throwing an error, add the `throws` keyword to the function header

Reply:
done!